weatherman.py

```diff
@@ -1,214 +1,206 @@
+"""
+
+import csv
+import glob
+
+max_temp_dict = {}  # Monthly maximum temp will store e.g 1996:45
+min_temp_dict = {}  # Monthly minimum temp will store e.g 1996:3
+min_humidity_dict = {}  # Monthly minimum humidity will store e.g 1996:31
+max_humidity_dict = {}  # Monthly maximum temp will store e.g 1996:100
+complete_date_dict = {}  # Complete date will store e.g 1996:1996-2-3
+
+
+def annual_report():
+    """
+    It will show the annual report to the user
+
+    """
+
+    print(str("Year").rjust(7),
+          str("MAXTemp").rjust(7),
+          str("MINTemp").rjust(7),
+          str("MAXHumidity").rjust(7),
+          str("MINHumidity").rjust(7),
+          "[Report#1][Annual Report]".rjust(8)
+          )
+    print("-"*80)
+    # iterate over dictionary key @Year e.g 1996 and display it's value.Uses same key for all dictionaries
+    for i in complete_date_dict.keys():
+        print(str(i).rjust(6),
+              str(max_temp_dict[i]).rjust(6),
+              str(min_temp_dict[i]).rjust(6),
+              str(max_humidity_dict[i]).rjust(8),
+              str(min_humidity_dict[i]).rjust(9)
+              )
+
+
+def year_hottest_days():
+    """
+    it will show the the Hottest days with year and complete date
+
+    """
+    print(str("Year").rjust(4),
+          str("Date").rjust(10),
+          str("Temp").rjust(11),
+          "[Report#2][Hottest Days Report]".rjust(8)
+          )
+    print("-"*80)
+    # iterate over dictionary key @Year e.g 1996 and display it's value.Uses same key for two dictionaries
+    # e.g 1996:1996-12-1 and 1996:49
+    for i in complete_date_dict.keys():
+        print(str(i).rjust(4),
+              str(complete_date_dict[i]).rjust(12),
+              str(max_temp_dict[i]).rjust(8)
+              )
+
+
+def max_temp_func(temp_list, date_list):
+    """
+    This func will return the Monthly Maximum Temperature and Date
+    :param temp_list: one month temperatures
+    :param date_list: temperatures with complete date
+    :return: maximum temperature with complete date in list
+    """
+    # MAX TEMPERATURE
+    if len(temp_list) == 0:
+        return 0
+    if len(temp_list) == 1:
+        return list(zip(temp_list, date_list))
+    return list(max(zip(temp_list, date_list)))
+
+
+def min_func(list_1):
+    """
+    :param list_1: list to apply minimum function
+    :return: string minimum number
+    """
+    if len(list_1) == 0:
+        return 100
+    if len(list_1) == 1:
+        return list_1
+    minimum = min(map(int,list_1))
+    return str(minimum)
+
+
+def max_finder_func(list_1):
+    """
+
+    :param list_1: list to apply max function
+    :return: string minimum number
+    """
+    if len(list_1) == 0:
+        return 0
+    if len(list_1) == 1:
+        return list_1
+    maximum = max(map(int, list_1))
+    return str(maximum)
+
+
+def main_function():
+    """
+    This is Main function
+
+    """
+    files = glob.glob("*.txt")
+    for file in files:
+        MAXTemp = []

```

Comment:
variable name are always written in small letters **max_temp**, unless some sort of global or settings variable written in all caps.